test/post-task.ts

```diff
@@ -96,19 +96,48 @@ test('postTask() validates abortSignal', async ({ rejects }) => {
     /signal argument must be an object/);
 });
 
-test('Piscina emits drain', async ({ ok }) => {
+test('Piscina emits drain', async ({ ok, notOk }) => {
   const pool = new Piscina({
     filename: resolve(__dirname, 'fixtures/eval.js')
   });
 
   let drained = false;
+  let needsDrain = true;
   pool.on('drain', () => {
     drained = true;
+    needsDrain = pool.needsDrain;
   });
 
-  await Promise.all([pool.runTask('123'), pool.runTask('123')]);
+  await Promise.all([pool.run('123'), pool.run('123')]);
 
   ok(drained);
+  notOk(needsDrain);
+});
+
+test('Piscina exposes/emits needsDrain to true when capacity is exceeded', async ({ ok }) => {
+  const pool = new Piscina({
+    filename: resolve(__dirname, 'fixtures/eval.js'),
+    maxQueue: 3,
+    maxThreads: 1
+  });
+
+  let triggered = false;
+  let drained = false;
+  pool.once('drain', () => {
+    drained = true;
+  });
+  pool.once('needsDrain', () => {
+    triggered = true;
+  });
+
+  pool.run('123');
```

Comment:
Is it reliable? Is there a chance for the first '123' to be executed before the last `pool.run('123')` so pool.once('needsDrain') won't be called?

Reply:
It should, as we are limiting the threads to 1, so we just paid that overhead once. Then by the fourth one, the first is about or already finished, so the `needsDrain` is already triggered by then, the same as `drain`.

It can sound tricky, but it works 😅 

Do you have a suggestion of how to improve it?